sympy/geometry/line.py

```diff
@@ -2121,10 +2121,12 @@ def __new__(cls, p1, p2, **kwargs):
         p2 = Point(p2, dim=2)
         if p1 == p2:
             return p1
-        if (p1.x > p2.x) == True:
-            p1, p2 = p2, p1
-        elif (p1.x == p2.x) == True and (p1.y > p2.y) == True:
-            p1, p2 = p2, p1
+        dir = kwargs.get('dir', False)
```

Comment:
The docstring should be edited to tell that a Segment can now be directed. Should the same thing be done for the 3D version of Segment?

Did you consider using the keyword 'evaluate' instead of 'dir'? That's a common way to get an object returned with the arguments in the order given, e.g. `Mul(x, 2, evaluate=False).args == (x, 2)`